### DISCUSSION TITLE
subset cells by gene scores

### DISCUSSION THREAD

**Discussion by breannesparta** [GreenleafLab/ArchR#835] [Questions / Documentation]
Hi, I have a project where rather than performing unsupervised cell clustering, I'd like to subset my data based on the accessibility of two marker genes. Specifically, after performing QC filtering and computing the GeneScoreMatrix, I'd like to first visualize the data using the gene scores of two genes, then create a filter to subset genes that meet some threshold criteria. However, I am uncertain about how to index the GeneScoreMatrix object using gene names. Would anyone have an example of how to do this?

Our idea was to first pull the GeneScoreMatrix using the getMatrixFromProject function, and then create two new vectors - one for each gene of interest, then visualize them using the ggPoint scatter function, and then create a filter that extracts cells that meet two GeneScoreMatrix criteria. 

Thanks!

**Comment by rcorces** [comment]
Just use `subsetArchRProject()` after figuring out which cells you want to keep.

**Reply by breannesparta** [reply]
ok, thank you. The core challenge we are struggling with is figuring out how to call data from the GeneScoreMatrix object using gene names.

**Reply by rcorces** [reply]
`getMatrixFromProject()` returns a `SummarizedExperiment`. `rowData` contains the gene names. Perhaps its worthwhile to review how to manipulate `SummarizedExperiment` objects.

```
> se <- getMatrixFromProject(ArchRProj = projHeme5, useMatrix = "GeneScoreMatrix")
ArchR logging to : ArchRLogs/ArchR-getMatrixFromProject-365f196d691d-Date-2021-06-22_Time-08-27-38.log
If there is an issue, please report to github with logFile!
2021-06-22 08:27:42 : Organizing colData, 0.079 mins elapsed.
2021-06-22 08:27:42 : Organizing rowData, 0.08 mins elapsed.
2021-06-22 08:27:42 : Organizing rowRanges, 0.08 mins elapsed.
2021-06-22 08:27:42 : Organizing Assays (1 of 1), 0.08 mins elapsed.
2021-06-22 08:27:43 : Constructing SummarizedExperiment, 0.084 mins elapsed.
2021-06-22 08:27:44 : Finished Matrix Creation, 0.101 mins elapsed.
> se
class: SummarizedExperiment 
dim: 23127 10251 
metadata(0):
assays(1): GeneScoreMatrix
rownames: NULL
rowData names(6): seqnames start ... name idx
colnames(10251): scATAC_BMMC_R1#TTATGTCAGTGATTAG-1 scATAC_BMMC_R1#AAGATAGTCACCGCGA-1 ...
  scATAC_PBMC_R1#TTCGTTACATTGAACC-1 scATAC_PBMC_R1#CGCTATCGTGAGGTCA-1
colData names(28): BlacklistRatio DoubletEnrichment ... ReadsInPeaks FRIP

> rowData(se)
DataFrame with 23127 rows and 6 columns
      seqnames     start       end  strand         name     idx
         <Rle>   <array>   <array> <array>      <array> <array>
1         chr1     11874     14409       1      DDX11L1       1
2         chr1     29961     14362       2       WASH7P       2
3         chr1     69091     70008       1        OR4F5       3
4         chr1    140566    134773       2    LOC729737       4
5         chr1    714068    700245       2 LOC100288069       5
...        ...       ...       ...     ...          ...     ...
23123     chrX 154299695 154351349       1        BRCC3     906
23124     chrX 154444701 154468098       1         VBP1     907
23125     chrX 154493852 154487526       2       RAB39B     908
23126     chrX 154563986 154505500       2        CLIC2     909
23127     chrX 154842622 154718673       2        TMLHE     910
```

**Reply by breannesparta** [reply]
great, thank you
